### PR TITLE
Boot/Resume Overhaul

### LINKED ISSUE #136: Closing Port closes urbit processes
When running Port on Windows, closing the program also closes any urbit processes it spawned.

**Expected Behavior**
Urbit processes should stay running after closing port (until we make a setting to change this).

**Environment:**
 - OS: Windows
 - Version: 10
 - Port Version: 1.4

### LINKED ISSUE #199: Windows fails to eject planet

### PR DESCRIPTION
**Pull request by latter-bolden**
closes #136
closes #199

Okay so I started by trying to get urbit processes to run in the background on Windows and realized I also wanted the ability to do the opposite on my Mac. That led me down a rabbit hole that resulted in refactoring a decent amount of `pier-service` and logic around the boot/resume lifecycle in the UI.

## Summary of Changes
- Setting for optionally keeping ships running in the background (enabled by default)
  - when enabled, uses detached processes to achieve parity on Windows
  - when disabled, adds cleanup phase (backend and UI) for stopping all running ships on app quit
- Pid tracking for all ships rather than just during initial boot
- Fixes Windows signal issue and defends against killing of pid's that no longer exist
  - resolves the "export while booting" problem
- Added option for killing Urbit processes via signal rather than loopback dojo command
  - when ships should not be run in the background, this allows us to quit the app much more quickly
  - Vere gracefully handles these signals
 - Rewritten initial boot recovery, handle cases more granularly for cleaner resumes
   - fixes unintentionally reusing keyfile on retries
- Account for ships that error outside of boot in `pier-service` and UI (we were previously swallowing these)
- Improved ship status tracking and cache invalidation in UI
- Trim some dead code and a handful of cleanliness changes (renames, style consistency, better debug logs)

<br/>
All that said, this definitely got a bit overzealous, so let me know if there are pieces you want me to revert.

## Remaining Work
I wanted to get this up, but I still need to do a few things before this is worth reviewing:
- Resume errors are now surfaced to the UI, but I need to fix their layout
- I'm dissatisfied with the status bloat, need to figure out if I can streamline boot logic to collapse the recovery/error distinction
- More cross platform testing
-  Tidy up — haven't done a pass for code style, straggling `console.log`s, etc.

### CODE REVIEW

File: src/background/services/pier-service.ts [hunk 1]
```diff
@@ -638,18 +687,31 @@ export class PierService {
         return joinPath(userData, 'piers')
     }
 
-    private async recoverBootingShips() {
+    private async recoverShips() {
         const bootingShips = await this.db.piers.asyncFind({ 
-            $and: [{ status: 'booting' }, { bootProcessId: { $exists: true }}] 
+            $and: [ { $or: [{ status: 'booting' }, { status: 'resuming' }]}, { pid: { $exists: true }}]
```
Comment (arthyn) [inline src/background/services/pier-service.ts]: You have 'resuming' here but should that be starting?

**Review by arthyn — APPROVED** [review]
This is really awesome we desperately needed this haha. It looks good so far!

**Comment by latter-bolden** [comment]
@arthyn okay i still need to do cross platform testing, but this should be ready for review

I did change things up a bit from the first push. Instead of multiplying the statuses for all of the initial boot conditions, there's now a new pier field for tracking these (`startupPhase`) and the core statuses are reduced to _stopped, booting, running_ and _errored_.

I think this simplifies the mental model and better encapsulates the issues around initial boot since only `pier-service` should really care about those. It also made it straightforward for me to collapse boot and resume into a single operation. A big win here is that we can keep a lot of this complexity out of the interface code while also limiting the amount of damage the UI could unintentionally inflict on ships.

The downside is this change isn't backwards compatible so i'd have to whip up a migration flow before we can release. I don't think that would be too time consuming tho

Lemme know whatcha think

**Review by arthyn — APPROVED** [review]
This looks good. Even though you set the default for keep ships running to true mine defaulted to off.

Just need a migration script now

**Comment by latter-bolden** [comment]
This is good for final review. 

Was toying with a full version checking migration pipeline kinda deal, but it came out feeling like overkill for this. Instead, we just check to see if there are any schema differences we care about on every startup.

Needs more cross platform poking, but I'll do that once this is merged and we have a potential release artifact to work with.

**Review by arthyn — APPROVED** [review]
Looks good, you think we're ready for release?

**Pull request merged**